### PR TITLE
pubsub: change Topic.Close() to Topic.Shutdown(ctx) 

### LINKED ISSUE #883: pubsub: Topic.Shutdown(ctx)
`Subscription.Close()` is being converted to `Shutdown(ctx)`. It seems inconsistent not to do the same for `Topic.Close()`.

### PR DESCRIPTION
**Pull request by ijt**
Fixes #883

### CODE REVIEW

File: internal/pubsub/pub_test.go [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+func (t *funcTopic) Close() error {
+	return nil
+}
+
+func (s *funcTopic) IsRetryable(error) bool { return false }
+
+func (s *funcTopic) As(i interface{}) bool { return false }
+
+func TestTopicShutdownCanBeCanceledEvenWithHangingSend(t *testing.T) {
```
Comment (vangent) [inline internal/pubsub/pub_test.go]: Consider adding this test in a separate PR, it is not related to the rename.

File: internal/pubsub/pubsub.go [hunk 2]
```diff
@@ -88,14 +91,23 @@ func (t *Topic) Send(ctx context.Context, m *Message) error {
...
+// Shutdown flushes pending message sends and disconnects the Topic.
 // It only returns after all pending messages have been sent.
-func (t *Topic) Close() error {
+func (t *Topic) Shutdown(ctx context.Context) error {
 	t.mu.Lock()
 	t.err = errors.New("pubsub: Topic closed")
 	t.mu.Unlock()
-	t.batcher.Shutdown()
-	return nil
+	c := make(chan struct{})
```
Comment (vangent) [inline internal/pubsub/pubsub.go]: Same here, this isn't related to the rename is it?

File: internal/pubsub/pub_test.go (hunk 1 above)
Comment (ijt) [reply internal/pubsub/pub_test.go]: The rename isn't just a rename. The new name reflects new functionality, and this test demonstrates the new functionality.

File: internal/pubsub/pubsub.go (hunk 2 above)
Comment (ijt) [reply internal/pubsub/pubsub.go]: Same answer.

**Pull request merged**